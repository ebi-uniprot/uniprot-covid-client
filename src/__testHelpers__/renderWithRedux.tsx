/* eslint-disable */
import React from 'react';
import { createStore, applyMiddleware } from 'redux';
import { render } from '@testing-library/react';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { createMemoryHistory, MemoryHistory } from 'history';
import rootReducer from '../state/rootReducer';
import { Router, Route } from 'react-router-dom';

type RenderOptions = {
  route?: string;
  history?: MemoryHistory<any>;
  path?: string;
  initialState?: any;
  store?: any;
};

const renderWithRedux = (
  ui: React.Component,
  {
    route = '',
    history = createMemoryHistory({ initialEntries: [route] }),
    path,
    initialState,
    store = createStore(rootReducer, initialState, applyMiddleware(thunk)),
  }: RenderOptions = {}
) => {
  return {
    ...render(
      <Provider store={store}>
        <Router history={history}>
          {path ? <Route path={path} render={() => ui} /> : ui}
        </Router>
      </Provider>
    ),
    store,
    history,
  };
};

export default renderWithRedux;
