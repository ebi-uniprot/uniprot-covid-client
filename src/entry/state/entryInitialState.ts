import {
  UniProtkbUIModel,
  UniProtkbInactiveEntryModel,
} from '../../model/uniprotkb/UniProtkbConverter';
import { LiteratureForProteinAPI } from '../../literature/types/LiteratureTypes';
import { Facet } from '../../types/responseTypes';

export type EntryState = {
  accession: string | null;
  isFetching: boolean;
  data: UniProtkbUIModel | UniProtkbInactiveEntryModel | null;
  publicationsData: {
    isFetching: boolean;
    data: LiteratureForProteinAPI[];
    facets: Facet[];
    nextUrl: string;
    total: number;
  };
};

const entryInitialState = {
  accession: null,
  isFetching: false,
  data: null,
  publicationsData: {
    isFetching: false,
    data: [],
    facets: [],
    nextUrl: '',
    total: 0,
  },
};

export default entryInitialState;
