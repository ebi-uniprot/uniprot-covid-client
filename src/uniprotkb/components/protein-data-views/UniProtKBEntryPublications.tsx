import React, { FC, Fragment } from 'react';
import { Publication, Loader, Message } from 'franklin-sites';
import useDataApi from '../../../shared/hooks/useDataApi';
import { getPublicationsURL } from '../../config/apiUrls';
import { MessageLevel } from '../../../messages/types/messagesTypes';
import { LiteratureAPI } from '../../types/literatureTypes';
import formatCitationData from '../../adapters/literatureConverter';

const UniProtKBEntryPublications: FC<{
  pubmedIds: string[];
}> = ({ pubmedIds }) => {
  const url = getPublicationsURL(pubmedIds);
  const { loading, data, status, error } = useDataApi(url);

  if (error) {
    return (
      <Message level={MessageLevel.FAILURE}>
        {status}: {error.message}
      </Message>
    );
  }

  if (loading || !data) {
    return <Loader />;
  }

  const { results }: { results: LiteratureAPI[] } = data;
  return (
    <Fragment>
      {results &&
        results
          .map(literatureItem => ({
            ...literatureItem,
            ...formatCitationData(literatureItem.citation),
          }))
          .map(({ citation, statistics, pubmedId, journalInfo }) => (
            <Publication
              title={citation.title}
              authors={citation.authors}
              key={`${citation.title}-${citation.citationType}-${citation.journal}`}
              pubmedId={pubmedId}
              statistics={statistics}
              journalInfo={journalInfo}
            />
          ))}
    </Fragment>
  );
};

export default UniProtKBEntryPublications;
