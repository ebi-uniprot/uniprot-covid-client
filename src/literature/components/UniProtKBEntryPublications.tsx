import React, { FC, Fragment } from 'react';
import { connect } from 'react-redux';
import { Publication } from 'franklin-sites';
import { UniProtkbUIModel } from '../../model/uniprotkb/UniProtkbConverter';
import { RootState } from '../../state/state-types';

const UniProtKBEntryPublications: FC<{
  pubmedIds: string[];
  entryData: UniProtkbUIModel | null;
}> = ({ pubmedIds, entryData }) => {
  if (!entryData) {
    return null;
  }
  const references =
    entryData.references &&
    entryData.references.filter(
      ({ citation }) =>
        citation.citationCrossReferences &&
        citation.citationCrossReferences.some(
          ({ id }) => id && pubmedIds.includes(id)
        )
    );
  return (
    <Fragment>
      {references &&
        references.map(({ citation }) => (
          <Publication
            title={citation.title}
            authors={citation.authors}
            key={`${citation.title}-${citation.citationType}-${citation.journal}`}
            journalInfo={{
              firstPage: citation.firstPage,
              journal: citation.journal,
              lastPage: citation.lastPage,
              publicationDate: citation.publicationDate,
              volume: citation.volume,
            }}
          />
        ))}
    </Fragment>
  );
};

const mapStateToProps = (state: RootState) => ({
  entryData: state.entry.data as UniProtkbUIModel | null,
});

export default connect(mapStateToProps)(UniProtKBEntryPublications);
