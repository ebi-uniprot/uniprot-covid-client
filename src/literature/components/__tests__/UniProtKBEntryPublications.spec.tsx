import React from 'react';
import renderWithRedux from '../../../__testHelpers__/renderWithRedux';
import UniProtKBEntryPublications from '../UniProtKBEntryPublications';
import entryData from '../../../model/__mocks__/entryModelData.json';
import entryInitialState from '../../../entry/state/entryInitialState';

describe('UniProtKBEntryPublications', () => {
  it('Should render', () => {
    const { asFragment } = renderWithRedux(
      <UniProtKBEntryPublications pubmedIds={['12345']} />,
      {
        initialState: {
          entry: { ...entryInitialState, accession: 'P05067', data: entryData },
        },
      }
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
