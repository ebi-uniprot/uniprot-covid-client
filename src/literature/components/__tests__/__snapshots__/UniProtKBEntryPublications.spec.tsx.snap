// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UniProtKBEntryPublications Should render 1`] = `
<DocumentFragment>
  <section
    class="publication"
  >
    <section
      class="publication__columns"
    >
      <section
        class="publication__columns__main"
      >
        <h5>
          Leo book tittle
        </h5>
        <section
          class="publication__authors"
        >
          <a
            href="/"
          >
            author Leo
          </a>
        </section>
      </section>
      <section
        class="publication__columns__side"
      >
        <section
          class="publication__columns__side__item"
        >
          <ul
            class="no-bullet"
          >
            <li>
              <small />
            </li>
          </ul>
        </section>
        <section
          class="publication__columns__side__item"
        >
          <section
            class="publication__statistics"
          />
        </section>
      </section>
    </section>
  </section>
  <section
    class="publication"
  >
    <section
      class="publication__columns"
    >
      <section
        class="publication__columns__main"
      >
        <h5>
          Leo book tittle
        </h5>
        <section
          class="publication__authors"
        >
          <a
            href="/"
          >
            author Leo
          </a>
        </section>
      </section>
      <section
        class="publication__columns__side"
      >
        <section
          class="publication__columns__side__item"
        >
          <ul
            class="no-bullet"
          >
            <li>
              <small />
            </li>
          </ul>
        </section>
        <section
          class="publication__columns__side__item"
        >
          <section
            class="publication__statistics"
          />
        </section>
      </section>
    </section>
  </section>
  <section
    class="publication"
  >
    <section
      class="publication__columns"
    >
      <section
        class="publication__columns__main"
      >
        <h5>
          Leo book tittle
        </h5>
        <section
          class="publication__authors"
        >
          <a
            href="/"
          >
            author Leo
          </a>
        </section>
      </section>
      <section
        class="publication__columns__side"
      >
        <section
          class="publication__columns__side__item"
        >
          <ul
            class="no-bullet"
          >
            <li>
              <small />
            </li>
          </ul>
        </section>
        <section
          class="publication__columns__side__item"
        >
          <section
            class="publication__statistics"
          />
        </section>
      </section>
    </section>
  </section>
  <section
    class="publication"
  >
    <section
      class="publication__columns"
    >
      <section
        class="publication__columns__main"
      >
        <h5>
          Leo book tittle
        </h5>
        <section
          class="publication__authors"
        >
          <a
            href="/"
          >
            author Leo
          </a>
        </section>
      </section>
      <section
        class="publication__columns__side"
      >
        <section
          class="publication__columns__side__item"
        >
          <ul
            class="no-bullet"
          >
            <li>
              <small />
            </li>
          </ul>
        </section>
        <section
          class="publication__columns__side__item"
        >
          <section
            class="publication__statistics"
          />
        </section>
      </section>
    </section>
  </section>
  <section
    class="publication"
  >
    <section
      class="publication__columns"
    >
      <section
        class="publication__columns__main"
      >
        <h5>
          Leo book tittle
        </h5>
        <section
          class="publication__authors"
        >
          <a
            href="/"
          >
            author Leo
          </a>
        </section>
      </section>
      <section
        class="publication__columns__side"
      >
        <section
          class="publication__columns__side__item"
        >
          <ul
            class="no-bullet"
          >
            <li>
              <small />
            </li>
          </ul>
        </section>
        <section
          class="publication__columns__side__item"
        >
          <section
            class="publication__statistics"
          />
        </section>
      </section>
    </section>
  </section>
  <section
    class="publication"
  >
    <section
      class="publication__columns"
    >
      <section
        class="publication__columns__main"
      >
        <h5>
          Leo book tittle
        </h5>
        <section
          class="publication__authors"
        >
          <a
            href="/"
          >
            author Leo
          </a>
        </section>
      </section>
      <section
        class="publication__columns__side"
      >
        <section
          class="publication__columns__side__item"
        >
          <ul
            class="no-bullet"
          >
            <li>
              <small />
            </li>
          </ul>
        </section>
        <section
          class="publication__columns__side__item"
        >
          <section
            class="publication__statistics"
          />
        </section>
      </section>
    </section>
  </section>
  <section
    class="publication"
  >
    <section
      class="publication__columns"
    >
      <section
        class="publication__columns__main"
      >
        <h5>
          Leo book tittle
        </h5>
        <section
          class="publication__authors"
        >
          <a
            href="/"
          >
            author Leo
          </a>
        </section>
      </section>
      <section
        class="publication__columns__side"
      >
        <section
          class="publication__columns__side__item"
        >
          <ul
            class="no-bullet"
          >
            <li>
              <small />
            </li>
          </ul>
        </section>
        <section
          class="publication__columns__side__item"
        >
          <section
            class="publication__statistics"
          />
        </section>
      </section>
    </section>
  </section>
</DocumentFragment>
`;
